src/main/scala/io/github/davidgregory084/TpolecatPlugin.scala

```diff
@@ -16,153 +16,127 @@
-            allScalacOptions
-              .filter(validFor(Version(maj, min, Try(patch.toLong).getOrElse(0))))
-          case (Some((maj, min)), _) =>
-            allScalacOptions
-              .filter(validFor(Version(maj, min, 0)))
-          case (None, _) =>
-            Nil
-        }
-
-      flags.map(_.name)
-}
-    val filterConsoleScalacOptions = { options: Seq[String] =>
-      options.filterNot(Set(
-        "-Werror",
-        "-Wdead-code",
-        "-Wunused:imports",
-        "-Ywarn-unused",
-        "-Ywarn-unused:imports",
-        "-Ywarn-unused-import",
-        "-Ywarn-dead-code",
-        "-Xfatal-warnings"
-      ))
+    val tpolecatConsoleOptionsFilter = { options: Set[ScalacOption] =>
+      options.filterNot(
+        ScalacOptions.privateWarnUnusedOptions ++
+          ScalacOptions.warnUnusedOptions ++
+          ScalacOptions.fatalWarningOptions +
+          ScalacOptions.privateWarnDeadCode +
+          ScalacOptions.warnDeadCode
+      )
     }
+
+    val tpolecatDefaultOptionsMode = settingKey[OptionsMode](
+      "The default mode to use for configuring scalac options via the sbt-tpolecat plugin."
+    )
+
+    val tpolecatOptionsMode = settingKey[OptionsMode](
+      "The mode to use for configuring scalac options via the sbt-tpolecat plugin."
+    )
+
+    val tpolecatDevModeEnvVar = settingKey[String](
+      "The environment variable to use to enable the sbt-tpolecat development mode."
+    )
+
+    val tpolecatCiModeEnvVar = settingKey[String](
+      "The environment variable to use to enable the sbt-tpolecat continuous integration mode."
+    )
+
+    val tpolecatReleaseModeEnvVar = settingKey[String](
+      "The environment variable to use to enable the sbt-tpolecat release mode."
+    )
+
+    val tpolecatDevModeOptions = settingKey[Set[ScalacOption]](
+      "The set of scalac options that will be applied by the sbt-tpolecat plugin in the development mode."
+    )
+
+    val tpolecatCiModeOptions = settingKey[Set[ScalacOption]](
+      "The set of scalac options that will be applied by the sbt-tpolecat plugin in the continuous integration mode."
+    )
+
+    val tpolecatReleaseModeOptions = settingKey[Set[ScalacOption]](
+      "The set of scalac options that will be applied by the sbt-tpolecat plugin in the release mode."
+    )
+
+    val tpolecatScalacOptions = settingKey[Set[ScalacOption]](
+      "The set of scalac options that will be applied by the sbt-tpolecat plugin."
+    )
   }
 
   import autoImport._
 
+  val commandAliases =
+    addCommandAlias(
+      "tpolecatDevMode",
+      "set tpolecatOptionsMode := _root_.io.github.davidgregory084.DevMode"
+    ) ++
+      addCommandAlias(
+        "tpolecatCiMode",
+        "set tpolecatOptionsMode := _root_.io.github.davidgregory084.CiMode"
+      ) ++
+      addCommandAlias(
+        "tpolecatReleaseMode",
+        "set tpolecatOptionsMode := _root_.io.github.davidgregory084.ReleaseMode"
```

Comment:
Hmm I tried this but unfortunately I didn't have any great success with it - I found that whichever mode that you set only lasted until you ran another command that used `tpolecatScalacOptions`, at which point the `tpolecatDefaultOptionsMode` was restored.
Since [sbt-scoverage](https://github.com/scoverage/sbt-scoverage/blob/main/src/main/scala/scoverage/ScoverageSbtPlugin.scala#L54=) uses a similar approach I may stick with this.